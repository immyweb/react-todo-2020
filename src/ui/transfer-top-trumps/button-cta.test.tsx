import * as React from 'react';
import { shallow } from 'enzyme';

import ButtonCta from './button-cta';

function shallowRender(props?: any) {
  return shallow(
    <ButtonCta
      bkgndTop={'#000000'}
      bkgndBottom={'#ffffff'}
      iconName={'cross-solid'}
      onClick={() => {}}
      {...props}
    />,
  );
}

describe('<ButtonCta />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('can change size, border width and icon size', () => {
    const wrapper = shallowRender({ size: 20, borderWidth: 1, iconSize: 6 });
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick is called when button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onClick: fn });
    wrapper.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('can be disabled', () => {
    const wrapper = shallowRender({ disabled: true });
    expect(wrapper.props().disabled).toEqual(true);
  });
});
