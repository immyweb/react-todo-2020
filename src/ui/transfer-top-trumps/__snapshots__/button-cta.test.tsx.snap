// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCta /> can change size, border width and icon size 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonCta
    bkgndBottom="#ffffff"
    bkgndTop="#000000"
    borderWidth={1}
    iconName="cross-solid"
    iconSize={6}
    onClick={[Function]}
    size={20}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bkgndBottom": "#ffffff",
      "bkgndTop": "#000000",
      "borderWidth": 1,
      "children": <Icon
        iconColour="#ffffff"
        iconName="cross-solid"
        iconSize={6}
      />,
      "className": undefined,
      "disabled": undefined,
      "onClick": [Function],
      "size": 20,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "iconColour": "#ffffff",
        "iconName": "cross-solid",
        "iconSize": 6,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bkgndBottom": "#ffffff",
        "bkgndTop": "#000000",
        "borderWidth": 1,
        "children": <Icon
          iconColour="#ffffff"
          iconName="cross-solid"
          iconSize={6}
        />,
        "className": undefined,
        "disabled": undefined,
        "onClick": [Function],
        "size": 20,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "iconColour": "#ffffff",
          "iconName": "cross-solid",
          "iconSize": 6,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<ButtonCta /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonCta
    bkgndBottom="#ffffff"
    bkgndTop="#000000"
    iconName="cross-solid"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bkgndBottom": "#ffffff",
      "bkgndTop": "#000000",
      "borderWidth": undefined,
      "children": <Icon
        iconColour="#ffffff"
        iconName="cross-solid"
        iconSize={32}
      />,
      "className": undefined,
      "disabled": undefined,
      "onClick": [Function],
      "size": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "iconColour": "#ffffff",
        "iconName": "cross-solid",
        "iconSize": 32,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bkgndBottom": "#ffffff",
        "bkgndTop": "#000000",
        "borderWidth": undefined,
        "children": <Icon
          iconColour="#ffffff"
          iconName="cross-solid"
          iconSize={32}
        />,
        "className": undefined,
        "disabled": undefined,
        "onClick": [Function],
        "size": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "iconColour": "#ffffff",
          "iconName": "cross-solid",
          "iconSize": 32,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
